Spring-Boot/Spring-Boot-Exam/src/main/java/tw/waterball/judgegirl/springboot/exam/view/ExamOverview.java

```diff
@@ -22,15 +23,19 @@
     private List<QuestionOverview> questionViews;
     private int totalScore;
 
-    public static ExamOverview toViewModel(Exam exam) {
-        return ExamOverview.builder()
+    public static ExamOverview toViewModel(Exam exam, List<ProblemView> problemViews) {
+        ExamOverview examOverview = ExamOverview.builder()
                 .id(exam.getId())
                 .name(exam.getName())
                 .startTime(exam.getStartTime())
                 .endTime(exam.getEndTime())
                 .description(exam.getDescription())
-                .questionViews(exam.getQuestions().stream().map(QuestionOverview::toViewModel).collect(Collectors.toList()))
+                .questionViews(new ArrayList<>())
                 .totalScore(exam.getQuestions().stream().mapToInt(question -> question.getScore()).sum())
                 .build();
+        for (int i = 0; i < exam.getQuestions().size(); i++) {
+            examOverview.questionViews.add(QuestionOverview.toViewModel(exam.getQuestions().get(i), problemViews.get(i)));
+        }
+        return examOverview;

```

Comment:
Don't compromise to our _elegance_ :)

Do the followings to save the code:

1. Add the below method into the class `tw.waterball.judgegirl.commons.utils.StreamUtils`

```java
public static <T, L, R> List<T> zipToList(List<L> left, List<R> right,
                                              BiFunction<L, R, T> zipAndMap) {
        return IntStream.range(0, left.size())
                .mapToObj(i -> zipAndMap.apply(left.get(i), right.get(i)))
                .collect(Collectors.toList());
    }
```

2. Improve the code by:

`.questionViews(zipToList(exam.getQuestions(), problemViews, QuestionOverview:: toViewModel))`

see, it's elegant right?